CMakeLists.txt

```diff
@@ -562,12 +562,53 @@ set (gmic_qt_FORMS
 
 if(ENABLE_DYNAMIC_LINKING)
   set(CMAKE_SKIP_RPATH TRUE)
-  set(gmic_qt_LIBRARIES
-    ${gmic_qt_LIBRARIES}
-    "gmic"
+  # G'MIC-Qt needs visibility into the private symbols defined
+  # by the gmic.cpp plugin. However, this is only possible
+  # if the library is static OR if it's dynamic and built by
+  # a compiler that supports .so-style exports.
+  if (TARGET libgmicstatic OR MSVC OR NOT ENABLE_SYSTEM_GMIC)
+    set(gmic_qt_LIBRARIES
+      ${gmic_qt_LIBRARIES}
+      libgmicstatic
+    )
+  elseif(TARGET libgmic)
+    set(gmic_qt_LIBRARIES
+      ${gmic_qt_LIBRARIES}
+      libgmic
+    )
+  elseif(GMIC_LIB_PATH)
+    set(gmic_qt_LIBRARIES
+      ${gmic_qt_LIBRARIES}
+      "gmic"
     )
+  else()
+    message(FATAL_ERROR "No G'MIC library is available for linking. Please build libgmic as a static library.")
+  endif()
   if (NOT ENABLE_SYSTEM_GMIC)
-    link_directories(${GMIC_LIB_PATH})
+    if (GMIC_LIB_PATH)
+      link_directories(${GMIC_LIB_PATH})
+      # Inject the G'MIC CImg plugin.
+      include_directories(../src)
+    else()
+      # Mimic an external G'MIC library build for catching link ABI errors.
+      add_library(libgmicstatic STATIC ../src/gmic.cpp)
+      target_include_directories(libgmicstatic PUBLIC ../src)
+      # We need internal access into the gmic-core API.
+      target_compile_definitions(libgmicstatic PUBLIC gmic_core)
+      set_target_properties(libgmicstatic
+        PROPERTIES
+          AUTOMOC OFF
+      )
+      target_link_libraries(libgmicstatic PUBLIC
+        ${PNG_LIBRARIES}
+        ${FFTW3_LIBRARIES}
+        ${ZLIB_LIBRARIES}
+        ${CURL_LIBRARIES}
+        ${EXTRA_LIBRARIES})
```

Comment:
I wonder if we shipped the GMic CMake files in the tarball, we could just rely on the targets that are already defined there.

Reply:
@dtschump has actually removed CMake support in G'MIC, so using them will court a second PR to remove them altogether here in the future.

Reply:
> @dtschump has actually removed CMake support in G'MIC, so using them will court a second PR to remove them altogether here in the future.

Not completely true. I just don't want to maintain it (I am not interested in this build system). Last time I removed the `CMakeLists` from the official tarball was because it was not working anymore due to the changes I made on the source code. I'd be happy to put it back if I'm sure it works as expected (i.e. just as the  `Makefile` does). I don't want to get negative feedback from users (that I couldn't handle) because of a non-functional `CMakeLists`.

Reply:
@dtschump by removing (support for) CMake, you also cease to support non-Unix operating systems like Windows. This means that to compile, you need a full blown MSYS shell or at the very least a MinGW toolchain on your path.

Moreover, the same applies with the recent translations regeneration scripts. These hard depend on a Bourne shell being available, which is necessary to trigger the regeneration of the filter translations.

Reply:
@amyspark , as I said before, I'm ok to put a `CMakeList` back in the source tarball if I'm sure it's not broken and that G'MIC compiles just fine with it. It's just a matter of having a maintainer who takes care of translating any changes in the `Makefile` into the `CMakeList`.
Concerning the translations, we also put all the necessary files in the provided source tarball to compile G'MIC: 
- Current stable : https://gmic.eu/files/source/gmic_3.2.1.tar.gz
- Current development : https://gmic.eu/files/prerelease/gmic_prerelease.tar.gz

In any case, we consider the tarball file as the "full" sources of G'MIC. The git repo is just a working space to develop G'MIC, it's not intended to be a "ready-as-it-is" version of the sources.
If you want to compile G'MIC for Krita, then it would be better to fetch the G'MIC sources from the tarball rather than from the git repo.